src/components/button/button.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Button as ButtonPaper } from 'react-native-paper';
+import { IconSource } from 'react-native-paper/lib/typescript/components/Icon';
+
+import { AppColor, ButtonType } from 'src/common/enums';
+import styles from './styles';
+
+type IButtonProps = {
+  children: Element;
+  type: ButtonType;
+  onPress: () => void;
+  icon?: IconSource;
+};
+
+const Button: React.FC<IButtonProps> = ({ children, type, onPress, icon }) => {
+  return (
+    <ButtonPaper
+      uppercase={false}
+      style={{ ...styles.global, ...styles[type] }}
+      onPress={onPress}
+      mode={type}
+      icon={icon}
+      color={AppColor.ACCENT}
+    >
+      {children}
+    </ButtonPaper>
+  );
+};
```

Comment:
Let's stick to existing naming and just passthrough props which we don't need to use.

`color={AppColor.ACCENT}` -- why using the same color for all variants?

```suggestion
const Button: React.FC<IButtonProps> = ({ children, mode, ...buttonProps }) => {
  return (
    <ButtonPaper
      uppercase={false}
      style={{ ...styles.global, ...styles[mode] }}
      mode={mode}
      color={AppColor.ACCENT}
      {...buttonProps}
    >
      {children}
    </ButtonPaper>
  );
};
```

Reply:
Same color because it sets this color to background or text color depending on button mode. If the problem is that we can't pass another color, I'll change it